src/login/loginctl.c

```diff
@@ -497,102 +492,193 @@ static int print_session_status_info(sd_bus *bus, const char *path, bool *new_li
-                       FORMAT_TIMESTAMP(i.timestamp.realtime),
-                       FORMAT_TIMESTAMP_RELATIVE_MONOTONIC(i.timestamp.monotonic));
+        r = table_add_many(table,
+                           TABLE_FIELD, "State",
+                           TABLE_SET_ALIGN_PERCENT, 100,
+                           TABLE_STRING, i.state,
+                           TABLE_SET_ALIGN_PERCENT, 0);
+        if (r < 0)
+                return table_log_add_error(r);
 
         if (i.leader > 0) {
-                _cleanup_free_ char *t = NULL;
+                _cleanup_free_ char *name = NULL;
 
-                printf("\t  Leader: " PID_FMT, i.leader);
+                (void) get_process_comm(i.leader, &name);
 
-                (void) get_process_comm(i.leader, &t);
-                if (t)
-                        printf(" (%s)", t);
+                r = table_add_many(table,
+                                   TABLE_FIELD, "Leader",
+                                   TABLE_SET_ALIGN_PERCENT, 100);
+                if (r < 0)
+                        return table_log_add_error(r);
+
+                r = table_add_cell_stringf(table, &cell, PID_FMT "%s%s%s",
+                                           i.leader,
+                                           !isempty(name) ? " (" : "",
+                                           strempty(name),
+                                           !isempty(name) ? ")" : "");
+                if (r < 0)
+                        return table_log_add_error(r);
 
-                printf("\n");
+                (void) table_set_align_percent(table, cell, 0);
         }
 
-        if (!isempty(i.seat)) {
-                printf("\t    Seat: %s", i.seat);
+        r = table_add_many(table,
+                           TABLE_FIELD, "Seat",
+                           TABLE_SET_ALIGN_PERCENT, 100);
+        if (r < 0)
+                return table_log_add_error(r);
 
-                if (i.vtnr > 0)
-                        printf("; vc%u", i.vtnr);
+        if (i.vtnr > 0)
+                r = table_add_cell_stringf(table, &cell, "%s; vc%u", i.seat, i.vtnr);
+        else
+                r = table_add_cell(table, &cell, TABLE_STRING, i.seat);
+        if (r < 0)
+                return table_log_add_error(r);
 
-                printf("\n");
-        }
+        (void) table_set_align_percent(table, cell, 0);
 
         if (i.tty)
-                printf("\t     TTY: %s\n", i.tty);
+                r = table_add_many(table,
+                                   TABLE_FIELD, "TTY",
+                                   TABLE_SET_ALIGN_PERCENT, 100,
+                                   TABLE_STRING, i.tty,
+                                   TABLE_SET_ALIGN_PERCENT, 0);
         else if (i.display)
-                printf("\t Display: %s\n", i.display);
+                r = table_add_many(table,
+                                   TABLE_FIELD, "Display",
+                                   TABLE_SET_ALIGN_PERCENT, 100,
```

Comment:
hmm, these settings are automatically copied from the preceeding row, same cell, and thus you only really need to set them once, but not anymore after that.

Reply:
Indeed. And in `table_new_vertical` we already set the correct alignment. Will update.